accepted/2020/reproducible-builds.md

```diff
@@ -0,0 +1,238 @@
+- The validation tool must support public sources for sources and symbols.
+- The validation tool must be easy to use: single command the user invokes.
+
+**P1**
+
+- The validation tool should support non-public source and symbol locations given credentials.
+
+**P2**
+
+- The validation tool should support F#
+- The validation tool should support any .NET SDK tool which creates or modifies DLL / EXEs
+
+**Future**
+
+- Support for IL weavers
+
+### Non-Goals
+
+- Evaluating provenance of embedded resources, such as .resx or .baml.
+- Support for modification or patching during validation, to create a runnable binary that's functionally different from the original.
+- Support for exporting a rebuilt artifact
+- Supporting the tool being used to validate binaries in an environment different than what they were produced on. 
+- Validation of transitive dependency chain
+- Validation of existing packages. The feature requires new metadata added to the pdb.
+
+## Design
+
+The user experience is delivered through a .NET CLI tool, `dotnet validate ThePackage.nupkg` or `dotnet validate TheFile.dll`. The tool will be responsible for obtaining symbols from well-known symbol servers (NuGet public symbols, MSDL) and whatever the user's configured [symbol path](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/symbol-path) is.
+
+To manually validate a NuGet package, they can download it from NuGet.org or by using a `dotnet nuget download <packageId>` CLI command.
+
+Source files generated as part of the build pipeline are supported (eg. Source Generators, Code Behind) when `EmbedUntrackedSources` or `EmbedAllSources` is set to `true`. The generated sources are stored in the PDB file and are thus available for the validation steps. Source generators and other tooling are not re-executed, rather, the sources are extracted from the PDB and reused as input.
+
+IL weavers and trimmers won't be supported by the first version of the tool. We don't anticipate trimming is as relevant for libraries as we expect trimming to be primarily used by applications. The majority of libraries do not presently use weavers, so we are able to improve overall security even if we aren't able to validate those libraries. A future version would define a mechanism where weavers and trimmers would participate in the validation steps by storing their input settings in the pdb and providing a mechanism to obtain the specific version used, allowing it to be run again. The tools would also need to be deterministic with respect to their inputs.
+
+The tools require the following artifacts to work:
+
+- Portable PDBs (embedded is supported)
+- Binary references (referenced assemblies)
+- Source
+
+### PDBs
+
+PDBs are a critical component as they contain pointers to the original source, or embeded source. They also contain information about the binary references and compiler flags used during the original compilation.
```

Comment:
We should note the compiler version that is required for production of PDB here, and that they have to be portable PDB.